src/main/java/appeng/items/misc/WrappedGenericStack.java

```diff
@@ -142,6 +151,11 @@ public boolean overrideOtherStackedOnMe(ItemStack itemInSlot, ItemStack otherSta
                 long amount = unwrapAmount(itemInSlot);
                 long inserted = heldContainer.insert(what, amount, Actionable.MODULATE);
 
+                if (what instanceof AEFluidKey && System.currentTimeMillis() > lastFillSound + 50) {
+                    FluidSoundHelper.playFillSound(player, (AEFluidKey) what);
```

Comment:
Hi, thanks for the PR!

Looks good, however you should call `playFillSound` on the `heldContainer` to make it work with other key types too (e.g. chemicals in Applied Mekanistics).

Reply:
One question now that I think of it: under which circumstances is the sound played multiple times? Cause it seems like a bucket would only be filled once per click?

Reply:
When I was testing this, I was taking water from an interface (I also tried lava too), and for some reason, the `overrideStackOnMe` method was running twice when I picked up the water. This might be a separate bug, but I didn't want to mess anything up so I just worked around it.

Reply:
I think it would be better to figure out why it's called twice; I don't think it's supposed to happen.

Reply:
Could it be that it's called both on the client and the server? In which case checking if the player is a client player would probably do.

Reply:
You were right. I changed it to a client check and it worked perfectly.

Reply:
Ok, nice! Two things still: 😅
- Your check is going to crash on the server since `LocalPlayer` doesn't exist there. You should use `player.getLevel().isClientSide()` (or similar).
- You should also remove `FluidSoundHelper.playFillSound(player, (AEFluidKey) what);` since you're already playing the sound via the `heldContainer`.

Reply:
Fixed 😄